c2corg_ui/static/js/deletedocument.js

```diff
@@ -0,0 +1,98 @@
+goog.provide('app.deleteDocumentDirective');
+goog.provide('app.DeleteDocumentController');
+
+goog.require('app');
+
+/**
+ * This directive is used to manage the dialog to delete a document.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.deleteDocumentDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppDeleteDocumentController',
+    controllerAs: 'deldocCtrl',
+    templateUrl: '/static/partials/deletedocument.html',
+    bindToController: {
+      'module': '<'
+    }
+  };
+};
+
+app.module.directive('appDeleteDocument', app.deleteDocumentDirective);
+
+
+/**
+ * @param {appx.Document} documentData Data set as module value in the HTML.
+ * @param {app.Api} appApi appApi.
+ * @param {app.Alerts} appAlerts
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @param {ui.bootstrap.$modalStack} $uibModalStack $uibModalStack.
+ * @constructor
+ * @ngInject
+ */
+app.DeleteDocumentController = function(documentData, appApi, appAlerts, gettextCatalog, $uibModalStack) {
+
+  /**
+   * @type {appx.Document}
+   * @export
+   */
+  this.documentData = documentData;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.appApi_ = appApi;
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.appAlerts_ = appAlerts;
+
+  /**
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
+  /**
+   * @type {ui.bootstrap.$modalStack}
+   * @private
+   */
+  this.$uibModalStack_ = $uibModalStack;
+
+  /**
+   * @export
+   * @type {string}
+   */
+  this.module;
+};
+
+
+/**
+ * @export
+ */
+app.DeleteDocumentController.prototype.deleteDocument = function() {
+  this.appApi_.deleteDocument(this.documentData.document_id).then(
+      function(response) {
```

Comment:
We have discussed this with @tsauerwein about the documents merging (that does not redirect either).

It would make sense to redirect. both after merging or deleting, yes. On the other hand, as for now we have no way to display a confirmation message ("the document has successfully been merged or deleted") after a redirection. And keeping the document displayed could be useful if the user finally realizes they have forgotten to save some data. Finally it's a moderator-only feature and can be improved later if the moderators think it's better to redirect.